src/Symfony/Component/Console/Color.php

```diff
@@ -177,4 +181,32 @@ private function getSaturation(int $r, int $g, int $b): int
 
         return (int) $diff * 100 / $v;
     }
+
+    private function rgbToHex(string $color): string
+    {
+        if (!str_ends_with($color, ')')) {
+            throw new InvalidArgumentException('Invalid color; missing closing parenthesis at the end of RGB functional notation.');
+        }
+
+        $colorToExplode = substr(substr($color, 0, -1), 4);
+        $rgb = explode(',', $colorToExplode);
+
+        if (3 !== \count($rgb)) {
+            throw new InvalidArgumentException(sprintf('Invalid RGB functional notation; should be of the form "rgb(r, g, b)", got "%s".', $color));
+        }
```

Comment:
We could use a regex to validate and parse the string:

```suggestion
        if (!preg_match('/^rgb\(\s*(\d+)\s*,\s*(\d+)\s*,\s*(\d+)\s*\)$/', $color, $matches)) {
            throw new InvalidArgumentException(sprintf('Invalid RGB functional notation; should be of the form "rgb(r, g, b)", got "%s".', $color));
        }

        $rgb = array_slice($matches, 1);
```

Reply:
Suits me fine! It removes a lot of validation headaches. As always, thanks for your suggestion @derrabus! :pray: